src/coreclr/jit/compiler.cpp

```diff
@@ -5227,8 +5227,17 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
         if (JitConfig.JitDoReversePostOrderLayout())
         {
             auto lateLayoutPhase = [this] {
-                fgDoReversePostOrderLayout();
-                fgMoveColdBlocks();
+                // Skip preliminary reordering passes to create more work for 3-opt layout
+                if (compStressCompile(STRESS_THREE_OPT_LAYOUT, 10))
```

Comment:
I picked this frequency arbitrarily, though I suppose it doesn't have to be too high since this stress mode will always kick in for `JitStress=2`. I'm open to suggestions here.